AEPCore/Sources/rules/JSONRulesParser.swift

```diff
@@ -179,52 +191,104 @@ class JSONCondition: Codable {
-            if searchType == .ordered {
+            defer { semaphore.signal() }
+            switch searchType {
+            case .any:
                 for result in results {
-                    if result.count == -1 {
-                        // database error
+                    // If a database error is returned for any result, early exit and return the error value
+                    guard result.count != -1 else {
                         returnValue = -1
-                        semaphore.signal()
                         break
-                    } else if result.count == 0 {
-                        // quick exit on ordered searches if any result has a count == 0
-                        returnValue = 0
-                        semaphore.signal()
-                        break
-                    } else {
-                        returnValue = 1
                     }
+                    returnValue += result.count
                 }
-            } else {
+            case .mostRecent:
+                // Should be impossible to reach due to early exit guard
+                returnValue = -1
+            case .ordered:
                 for result in results {
-                    if result.count == -1 {
-                        // database error
+                    // If a database error is returned for any result, early exit and return the error value
+                    guard result.count != -1 else {
                         returnValue = -1
-                        semaphore.signal()
                         break
-                    } else {
-                        returnValue += result.count
                     }
+                    // Early exit on ordered searches if any event result returned no records
+                    if result.count == 0 {
+                        returnValue = 0
+                        break
+                    }
+                    returnValue = 1
                 }
             }
-
-            semaphore.signal()
         }
 
         semaphore.wait()
         return returnValue
     }
 
+    /// Returns the index of the most recent historical event based on occurrence timestamp.
+    /// If no events are found or an error occurs during lookup, the method returns `-1`.
+    /// If duplicate ``EventHistoryRequest``s are provided, the index of the first instance will be returned.
+    ///
+    /// - Parameter parameters: An array expected to contain, in order:
+    ///     - An ``ExtensionRuntime`` instance used to perform historical event lookups.
+    ///     - An array of ``EventHistoryRequest`` objects specifying the search criteria.
+    ///
+    /// - Returns: The index of the event with the most recent occurrence, or `-1` if none are found or there is an error.
+    func getMostRecentHistoricalEvent(parameters: [Any]?) -> Int {
+        guard let params = parameters,
+              params.count >= 2,
```

Comment:
iOS novice question: On L172, params passed into `getMostRecentHistoricalEvent` is an array of three elements, same as `getHistoricalEventCount`. Won't this check fail?